OpenRA.Mods.Common/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -101,7 +97,8 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 					new FacingInit(spawnFacing)
 				});
 
-				var exitCell = self.Location + exit.ExitCell;
+				var exit = SelectExit(self, producee, productionType);
+				var exitCell = exit != null ? self.Location + exit.Info.ExitCell : self.Location;
 				actor.QueueActivity(new Land(actor, Target.FromActor(self), WDist.Zero, WVec.Zero, info.Facing, clearCells: new CPos[1] { exitCell }));
```

Comment:
The delivery will behave incorrectly if `exitCell` changes between here and below. The `Land` logic will need to be updated to consider any possible exit, and the selection below updated to make sure it is consistent with the cleared cells.

This is non-trivial, and the main reason why this code assumes only a single exit!